src/wallet/wallet.cpp

```diff
@@ -245,7 +245,7 @@ bool CWallet::Unlock(const SecureString& strWalletPassphrase, bool anonymizeOnly
 {
     SecureString strWalletPassphraseFinal;
 
-    if (!IsLocked())
+    if (!IsLocked() && !fWalletUnlockAnonymizeOnly) // was already fully unlocked, not only for anonymization
```

Comment:
If the wallet is not locked but fWalletUnlockAnonymizeOnly == true, then the unlocking code will run on an already unlocked wallet.  Is that safe ?